api/client/create.go

```diff
@@ -99,6 +112,12 @@ func (cli *DockerCli) createContainer(config *runconfig.Config, hostConfig *runc
 		tag = tags.DefaultTag
 	}
 
+	if pull {
```

Comment:
Placing this here bypasses trust. Ensuring the latest image must be checked against trust. This may be as simple as moving this down and using the `trustedRef`.

cc @dmcgowan